args.h

```diff
@@ -175,28 +175,26 @@ typedef internal::EnumFlag Enum;
 typedef internal::IntFlag Int;
 typedef internal::BoolFlag Bool;
 
-// Constants used to make code using args framework more clear.
-// This may be possibly changed to an enum if there is need for more
-// "flag types" than REQUIRED and OPTIONAL.
-const bool REQUIRED = true;
-const bool OPTIONAL = false;
+// Denotes whether a flag is required or optional.
+// TODO: Very awkward name.
+enum Requiredness { REQUIRED, OPTIONAL };
```

Comment:
Me neither, but "Mandatory" is no good either. "Mandatory" is the same thing as "required" - I would expect `enum X { MANDATORY, OPTIONAL };`, not `enum Mandatory { X, OPTIONAL };` (it's a bit like `enum Red { RED, GREEN };`).

Maybe "Mandatoriness"? "Optionality"?